api-lib/circleAdmin.ts

```diff
@@ -0,0 +1,36 @@
+import assert from 'assert';
+
+import type {
+  VercelRequest,
+  VercelResponse,
+  VercelApiHandler,
+} from '@vercel/node';
+import { z } from 'zod';
+
+import { getUserFromAuthHeader } from './findUser';
+import { verifyHasuraAdminMiddleware } from './validate';
+
+const middleware =
+  (handler: VercelApiHandler) =>
+  async (req: VercelRequest, res: VercelResponse) => {
+    // skip auth check if no header is passed
+    // because we can assume the user is a hasura-admin
+    if (req.headers.authorization) {
+      const circleId = z.number().parse(req.body.input.circle_id);
+      const { role } = await getUserFromAuthHeader(
+        req.headers.authorization,
+        circleId
+      );
+      if (role !== 1) {
```

Comment:
small nit: I usually find logic patterns that are "if X then Y" easier to reason about and understand than "if not X, then not Y".

I can imagine it being nice to have a few helper functions here around the roles, so we could do something like this:

```
if role(admin) { 
do action
} else {
raise 401, unauthorized.
}
```

but don't need to refactor this right now, we can do it once we see a broader pattern clearly in a few of these PRs.

Reply:
sure, I bias towards "erroring early" but I definitely appreciate the desire to eliminate another thing that needs to be loaded into one's mental stack.

Reply:
i've created functions that describe the logic in words, since there are multiple checks now.